client/components/theme/index.jsx

```diff
@@ -62,10 +61,6 @@ const Theme = React.createClass( {
 		actionLabel: React.PropTypes.string
 	},
 
-	shouldComponentUpdate( nextProps ) {
-		return ! isEqual( nextProps.theme, this.props.theme );
```

Comment:
> I think we mostly need to make sure that buttonOptions don't change unnecessarily, and possibly by adding back shouldComponentUpdate, with a deep comparison (i.e. another isEqual) of buttonOptions alongside the one for theme, plus some comparisons for the stuff I mentioned above (screenshotClickUrl, onScreenshotClick, etc).

In this PR or a separate one? I don't want to hold this one up—on the other hand I'm loath to introduce a performance regression.